staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go

```diff
@@ -147,6 +147,24 @@ func Flatten(agg Aggregate) Aggregate {
 	return NewAggregate(result)
 }
 
+// UniqueAggregateWithCount removes duplicates from flattened Aggregate
+// and returns count for each unique error present in the Aggregate
+func UniqueAggregateWithCount(agg Aggregate) Aggregate {
+	if agg == nil || len(agg.Errors()) == 0 {
+		return agg
+	}
+	errs := Flatten(agg).Errors()
+	errorCountMap := map[string]int{}
+	for _, err := range errs {
+		errorCountMap[err.Error()]++
+	}
+	result := []error{}
```

Comment:
You know that `len(errorCountMap)` is # of keys that you have, so I would do `result := make([]error, 0, len(errorCountMap))`